consensus/state_processing/src/per_block_processing/process_operations.rs

```diff
@@ -450,59 +449,69 @@ pub fn apply_deposit<E: EthSpec>(
 
     if let Some(index) = validator_index {
         // [Modified in Electra:EIP7251]
-        if let Ok(pending_balance_deposits) = state.pending_balance_deposits_mut() {
-            pending_balance_deposits.push(PendingBalanceDeposit { index, amount })?;
-
-            let validator = state
-                .validators()
-                .get(index as usize)
-                .ok_or(BeaconStateError::UnknownValidator(index as usize))?;
-
-            if is_compounding_withdrawal_credential(deposit_data.withdrawal_credentials, spec)
-                && validator.has_eth1_withdrawal_credential(spec)
-                && is_valid_deposit_signature(&deposit_data, spec).is_ok()
-            {
-                state.switch_to_compounding_validator(index as usize, spec)?;
-            }
+        if let Ok(pending_deposits) = state.pending_deposits_mut() {
+            pending_deposits.push(PendingDeposit {
+                pubkey: deposit_data.pubkey,
+                withdrawal_credentials: deposit_data.withdrawal_credentials,
+                amount,
+                signature: deposit_data.signature,
+                slot: spec.genesis_slot, // Use `genesis_slot` to distinguish from a pending deposit request
+            })?;
         } else {
             // Update the existing validator balance.
             increase_balance(state, index as usize, amount)?;
         }
-    } else {
+    }
+    // New validator
+    else {
         // The signature should be checked for new validators. Return early for a bad
         // signature.
         if is_valid_deposit_signature(&deposit_data, spec).is_err() {
             return Ok(());
         }
 
-        let new_validator_index = state.validators().len();
-
         // [Modified in Electra:EIP7251]
-        let (effective_balance, state_balance) = if state.fork_name_unchecked() >= ForkName::Electra
-        {
-            (0, 0)
+        if state.fork_name_unchecked() >= ForkName::Electra {
+            let amount = 0;
+            // Create a new validator.
+            let mut validator = Validator {
+                pubkey: deposit_data.pubkey,
+                withdrawal_credentials: deposit_data.withdrawal_credentials,
+                activation_eligibility_epoch: spec.far_future_epoch,
+                activation_epoch: spec.far_future_epoch,
+                exit_epoch: spec.far_future_epoch,
+                withdrawable_epoch: spec.far_future_epoch,
+                effective_balance: 0,
+                slashed: false,
+            };
+            let max_effective_balance =
+                validator.get_max_effective_balance(spec, state.fork_name_unchecked());
+            validator.effective_balance = std::cmp::min(
+                amount.safe_sub(amount.safe_rem(spec.effective_balance_increment)?)?,
+                max_effective_balance,
```

Comment:
It's not unnecessary (this is logic for setting effective balance which is in the spec), but it IS a bug because amount has been hardcoded to zero. I think this bit of code is getting more and more difficult to manage as we add forks so I've created a PR to simplify this whole piece of code:

https://github.com/sigp/lighthouse/pull/6515

it needs a review to ensure we're compliant with the spec across all forks. But if we merge that into this PR it'll simplify this a ton.

Reply:
nvm this isn't a bug because the spec set `amount` to `0` much earlier as well.. but the simplification should still be done.

Reply:
Fixed with merging unstable and 37c7658 to get it more in line with the spec